commands/auth_utils.go

```diff
@@ -47,33 +54,52 @@ func init() {
 	SetUser(newUser)
 }
 
-func getDefaultConfigPath() string {
+func getDefaultConfigHomePath() string {
+	if p, ok := os.LookupEnv(strings.TrimPrefix(xdgConfigHomeVar, "$")); ok {
+		return p
+	}
+
+	return filepath.Join(currentUser.HomeDir, ".config")
+}
+
+func resolveLegacyConfigFilePath() string {
 	configPath := currentUser.HomeDir
 	// We use the existing $HOME/.mmctl file if it exists.
 	// If not, we try to read XDG_CONFIG_HOME and if we fail,
 	// we fallback to $HOME/.config/mmctl.
-	if _, err := os.Stat(filepath.Join(currentUser.HomeDir, "."+configFileName)); os.IsNotExist(err) {
+	if _, err := os.Stat(filepath.Join(currentUser.HomeDir, ".mmctl")); os.IsNotExist(err) {
 		if p, ok := os.LookupEnv(strings.TrimPrefix(xdgConfigHomeVar, "$")); ok {
 			configPath = p
 		} else {
 			configPath = filepath.Join(currentUser.HomeDir, ".config")
 		}
 	}
+
 	return configPath
 }
 
 func resolveConfigFilePath() string {
-	configPath := getDefaultConfigPath()
-	if p := viper.GetString("config-path"); p != xdgConfigHomeVar {
-		configPath = strings.Replace(viper.GetString("config-path"), userHomeVar, currentUser.HomeDir, 1)
-	}
+	// we warn users that config-path is deprecated
+	suppressWarnings := viper.GetBool("suppress-warnings")
```

Comment:
I am wondering if we could follow other commands and name this `--quiet`.

Reply:
`--quiet` may have some other context, it is mostly used for silencing all of the output but this is for rather suppressing warning messages.

Reply:
That would be `--silent` :p Just kidding, this is nothing blocking, just a thought.

Reply:
I get it covered :p https://github.com/mattermost/mmctl/pull/371